core/backend/src/IModelHost.ts

```diff
@@ -588,21 +608,48 @@ export class IModelHost {
   private static setupTileCache() {
     assert(undefined !== IModelHost.configuration);
     const config = IModelHost.configuration;
+    // eslint-disable-next-line deprecation/deprecation
     const service = config.tileCacheService;
+    const storage = config.tileCacheStorage;
     const credentials = config.tileCacheAzureCredentials;
 
-    if (!service && !credentials)
+    if(!service && !storage && !credentials) {
+      this.platform.setUseTileCache(true);
       return;
-
+    }
+    this.platform.setUseTileCache(false);
+    // eslint-disable-next-line deprecation/deprecation
     IModelHost.tileUploader = new CloudStorageTileUploader();
-
-    if (credentials && !service) {
-      IModelHost.tileCacheService = new AzureBlobStorage(credentials);
-    } else if (!credentials && service) {
-      IModelHost.tileCacheService = service;
-    } else {
-      throw new IModelError(BentleyStatus.ERROR, "Cannot use both Azure and custom cloud storage providers for tile cache.");
+    if(credentials) {
+      if(storage || service)
+        throw new IModelError(BentleyStatus.ERROR, "Cannot use both Azure and custom cloud storage providers for tile cache.");
+      this.setupAzureTileCache(credentials);
+    }
+    if(service) {
+      if(!storage)
+        Logger.logWarning(loggerCategory, "Using tileCacheService without tileCacheStorage is unsupported");
```

Comment:
Is this true? If so, have you broken existing apps that were using the now-deprecated but not removed APIs?

Reply:
In theory we have, but in practice not really.
There are two ways to configure a backend:
- by using Azure blob storage credentials - in that case everything will continue working as before because we initialize both `tileCacheService` and `tileCacheStorage` during `IModelHost.startup`.
- by providing your own implementation of `tileCacheService` - in that case the backend wouldn't be able to upload tiles because it now uses the new API to do that. I'm not aware of any backends that use this configuration, but even if someone does (maybe someone in China?), we decided the additional complexity of handling upload through both APIs would be worse than breaking a beta API that barely anyone ever used.

Reply:
As far as I know most apps provide `tileCacheAzureCredentials` in which case both service and storage are setup

Reply:
We promise stability between minor releases for all apps, not most apps.

Reply:
isn't the whole point of having a `@beta` API to be able to change it without strictly adhering to semver rules? We made sure to keep the RPC interface compatible with old frontends.

Reply:
Yes. It is not the API that concerns me. Technically preserving the API but breaking the implementation is much worse.
If you are breaking the implementation, why not just delete the API instead of deprecating it?

Reply:
I'd like to remove this one, but it's needed to get the tile cache URL used by older frontends. `@itwin/object-storage` doesn't support this workflow because it's incompatible with S3 so it wouldn't really be cloud-agnostic.

Reply:
Is there anyway to break the backend API but then leave the existing RPC interfaces the same? Since the main thing we care about for compatibility, with a beta API, is to not break existing frontends with a new backend version.

Reply:
that's what we did, if you configure the backend correctly it will remain compatible with old frontends. And if you use Azure credentials like most existing backends, your config will remain valid.